python/bindings/message_database.cpp

```diff
@@ -183,26 +184,48 @@ void init_common_message_database(nb::module_& m)
         .def("get_msg_def", nb::overload_cast<int32_t>(&PyMessageDatabase::GetMsgDef, nb::const_), "msg_id"_a)
         .def("get_enum_def", &PyMessageDatabase::GetEnumDefId, "enum_id"_a)
         .def("get_enum_def", &PyMessageDatabase::GetEnumDefName, "enum_name"_a)
-        .def_prop_ro("enums", &PyMessageDatabase::GetEnumsByNameDict);
+        .def_prop_ro("enums", &PyMessageDatabase::GetEnumsByNameDict)
+        .def_prop_ro("messages", &PyMessageDatabase::GetMessagesByNameDict);
 }
 
-PyMessageDatabase::PyMessageDatabase() { UpdatePythonEnums(); }
+PyMessageDatabase::PyMessageDatabase() { 
+    UpdatePythonEnums();
+    UpdateMessageTypes();
+}
 
 PyMessageDatabase::PyMessageDatabase(std::vector<MessageDefinition::ConstPtr> vMessageDefinitions_,
                                      std::vector<EnumDefinition::ConstPtr> vEnumDefinitions_)
     : MessageDatabase(std::move(vMessageDefinitions_), std::move(vEnumDefinitions_))
 {
     UpdatePythonEnums();
+    UpdateMessageTypes();
 }
 
-PyMessageDatabase::PyMessageDatabase(const MessageDatabase& message_db) noexcept : MessageDatabase(message_db) { UpdatePythonEnums(); }
+PyMessageDatabase::PyMessageDatabase(const MessageDatabase& message_db) noexcept : MessageDatabase(message_db) { 
+    UpdatePythonEnums();
+    UpdateMessageTypes();
+}
 
-PyMessageDatabase::PyMessageDatabase(const MessageDatabase&& message_db) noexcept : MessageDatabase(message_db) { UpdatePythonEnums(); }
+PyMessageDatabase::PyMessageDatabase(const MessageDatabase&& message_db) noexcept : MessageDatabase(message_db) { 
+    UpdatePythonEnums();
+    UpdateMessageTypes();
+}
 
 void PyMessageDatabase::GenerateMappings()
 {
     MessageDatabase::GenerateMappings();
     UpdatePythonEnums();
+    UpdateMessageTypes();
+}
+
+void cleanString(std::string& str)
+{
+    // Remove special characters from the string to make it a valid python attribute name
+    for (char& c : str)
+    {
+        if (c == '-' || c == '+' || c == '.' || c == '/' || c == '(' || c == ')') { c = '_'; }
```

Comment:
```suggestion
        if (!isalnum(c)) { c = '_'; }
```
I think it's simpler and safer to convert everything non-alphanumeric to an underscore.